webservices/rest/src/main/java/com/epam/rest/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.rest.dao;
+
+
+import com.epam.rest.domain.User;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    private final Map<Integer, User> storage = new HashMap<>();
+    private int counter = 0;
```

Comment:
I understand that it is just mock repository implementation, but you know that it is not thread-safe, right? 
What would be a way to make it safe, with minimum performance impact?